Controller/Transaction/AuthorizeOneclick.php

```diff
@@ -92,159 +102,223 @@ public function __construct(
             }
 
-            $inscription = $this->getOneclickInscriptionData($inscriptionId);
-            $username = $inscription->getUsername();
-            $tbkUser = $inscription->getTbkUser();
+            $requestMethod = $_SERVER['REQUEST_METHOD'];
+            $request = $requestMethod === 'POST' ? $_POST : $_GET;
 
-            $this->checkoutSession->restoreQuote();
+            $inscriptionId = intval($request['inscription']);
 
-            $quote = $this->cart->getQuote();
+            $this->log->logInfo('Autorizando transacción Oneclick.');
+            $this->log->logInfo('Request: method -> ' . $requestMethod);
+            $this->log->logInfo('Request: payload -> ' . json_encode($request));
 
-            $quote->getPayment()->importData(['method' => Oneclick::CODE]);
-            $quote->collectTotals();
-            $order = $this->getOrder();
-            $grandTotal = round($order->getGrandTotal());
+            return $this->handleOneclickRequest($inscriptionId);
+        } catch (Exception $e) {
+            return $this->handleException($e);
+        }
+    }
 
-            $quoteId = $quote->getId();
-            $orderId = $order->getId();
+    private function handleOneclickRequest(int $inscriptionId)
+    {
+        $inscription = $this->getOneclickInscriptionData($inscriptionId);
+        $username = $inscription->getUsername();
+        $tbkUser = $inscription->getTbkUser();
+
+        $this->checkoutSession->restoreQuote();
+        $quote = $this->cart->getQuote();
+        $quote->getPayment()->importData(['method' => Oneclick::CODE]);
+        $quote->collectTotals();
+        $quote->save();
+
+        $order = $this->getOrder($this->checkoutSession->getLastOrderId());
+        $orderId = $order->getId();
+        $grandTotal = round($order->getGrandTotal());
+
+        if ($this->checkTransactionIsAlreadyProcessed($orderId, $quote->getId())) {
+            return $this->handleTransactionAlreadyProcessed($orderId, $quote->getId());
+        }
+
+        $transbankSdkWebpay = new TransbankSdkWebpayRest($this->oneclickConfig);
 
-            $quote->save();
+        $buyOrder = "100000" . $orderId;
+        $childBuyOrder = "200000" . $orderId;
 
-            $transbankSdkWebpay = new TransbankSdkWebpayRest($this->oneclickConfig);
+        $details = [
+            [
+                "commerce_code" => $this->oneclickConfig['CHILD_COMMERCE_CODE'],
+                "buy_order" => $childBuyOrder,
+                "amount" => $grandTotal,
+                "installments_number" => 1
+            ]
+        ];
 
-            $buyOrder = "100000" . $orderId;
-            $childBuyOrder = "200000" . $orderId;
+        $authorizeResponse = $transbankSdkWebpay->authorizeTransaction($username, $tbkUser, $buyOrder, $details);
+
+        if (isset($authorizeResponse->details) && $authorizeResponse->details[0]->responseCode == 0) {
```

Comment:
as suggestion, this `0` can be a descriptive constant for a better readability

Reply:
Ok